UnitsNet/CustomCode/UnitParser.cs

```diff
@@ -54,11 +55,27 @@ internal static class UnitParser
 
             const string exponentialRegex = @"(?:[eE][-+]?\d+)?)";
 
+            // Special regex characters that need escaping
+            string[] regexSpecialCharacters = {"$","^","*","?"};
+            string[] unitAbbreviations = UnitSystem.GetCached(formatProvider)
+                .GetAllAbbreviations(unitType)
+                .OrderByDescending(s => s.Length) // Important to order by length -- if "m" is before "mm" and the input is "mm", it will match just "m" and throw invalid string error
+                .ToArray();
+            // Escape special regex characters
+            for (int i = 0; i < unitAbbreviations.Length; i++)
+            {
+                foreach (string specialCharacter in regexSpecialCharacters)
+                {
+                    unitAbbreviations[i] = unitAbbreviations[i].Replace(specialCharacter, @"\" + specialCharacter);
+                }
+            }
+            string unitsRegex = $"({String.Join("|", unitAbbreviations)})";
+
             string regexString = string.Format(@"(?:\s*(?<value>[-+]?{0}{1}{2}{3})?{4}{5}",
                 numRegex, // capture base (integral) Quantity value
                 exponentialRegex, // capture exponential (if any), end of Quantity capturing
                 @"\s?", // ignore whitespace (allows both "1kg", "1 kg")
-                @"(?<unit>[^\s\d,]+)", // capture Unit (non-whitespace) input
+                $@"(?<unit>{unitsRegex})", // capture Unit (non-whitespace) input
```

Comment:
The `@` is no longer needed. Update comment to something like `// capture Unit by a list of abbreviations`